### PR TITLE
feat: add PO

### PR DESCRIPTION
**Pull request by amaanq**
https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html

### CODE REVIEW

File: queries/po/highlights.scm
Comment (ObserverOfTime) [inline queries/po/highlights.scm:27]: I'm not keen on captures with three parts to be honest.

Comment (ObserverOfTime) [inline queries/po/highlights.scm:27]: I'm not keen on captures with three parts to be honest.

```suggestion
(comment (reference (text) @text.uri))
```

File: queries/po/highlights.scm
Comment (ObserverOfTime) [inline queries/po/highlights.scm:29]:
```suggestion
(comment (flag (text) @preproc))
```

File: queries/po/highlights.scm
Comment (ObserverOfTime) [inline queries/po/highlights.scm:20]: Missing the highlight for format strings.

```suggestion

(string_fragment) @string.special

```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/po/highlights.scm
Comment (clason) [reply queries/po/highlights.scm:27]: I don't think that's avoidable, though (especially not with our one-size-fits-none `@text` capture)...

And remember that there's builtin fallbacks, so you don't have to define a separate highlight group for it if you don't want to. There's a reason we only enforce top-level groups in CI.

Comment (clason) [reply queries/po/highlights.scm:27]: *Ceterum censeo*: We should replace `@text` with Helix' `@markup` hierarchy: https://docs.helix-editor.com/themes.html#syntax-highlighting

Comment (amaanq) [reply queries/po/highlights.scm:27]: It references a specific location in a file normally, e.g.
```
#: gnulib-lib/getopt.c:345
```

File: queries/po/highlights.scm
Comment (amaanq) [reply queries/po/highlights.scm:29]: Yeah I agree

File: queries/po/highlights.scm
Comment (amaanq) [reply queries/po/highlights.scm:20]: Do you mean like %s, %n?

File: queries/po/highlights.scm
Comment (ObserverOfTime) [reply queries/po/highlights.scm:27]: I know.

File: queries/po/highlights.scm
Comment (ObserverOfTime) [reply queries/po/highlights.scm:20]: Yep.

File: queries/po/highlights.scm
Comment (amaanq) [reply queries/po/highlights.scm:27]: > I know.

I think `@string.special` makes more sense due to the part after the file path doesn't really fit under `@text.uri`, but I'm open to adding both

File: queries/po/highlights.scm
Comment (amaanq) [reply queries/po/highlights.scm:20]: > Yep.

string_fragment captures any string content that isn't an escape sequence, so this doesn't fit. That being said, I could update the grammar to parse those as well, or wait until your format string injections is merged and inject here then.

Comment (ObserverOfTime) [reply queries/po/highlights.scm:20]: > string_fragment captures any string content that isn't an escape sequence, so this doesn't fit.

Oh, I didn't realise that. Can you try to capture them with `#match?` then?

File: queries/po/highlights.scm
Comment (ObserverOfTime) [reply queries/po/highlights.scm:27]: Well, it doesn't really fit `@string.special` either, but it's fine for now.

Maybe one day we'll rework all the highlight groups (including `@text` -> `@markup`).

Comment (amaanq) [reply queries/po/highlights.scm:27]: Mmm yeah I agree, let's rework this when that happens

File: queries/po/highlights.scm
Comment (amaanq) [reply queries/po/highlights.scm:20]: Mmm, sure but string_fragment is a token, how would I match inside that? This seems better suited for your injection (the printf format one)

The rule:

`string_fragment: _ => token.immediate(prec(1, /[^"\\]+/)),`

Comment (ObserverOfTime) [reply queries/po/highlights.scm:20]: I thought it matched words. Nevermind then.

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**
